coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,359 @@
+        Useless spacing affects the readability and maintainability of a code.
+        """
+        fix_suggestions = """
+        Removing the trailing spaces and the meaningless blank lines.
+        """
+
+
+@Spacing.subaspect
+class TrailingSpace:
+    """
+    Unnecessary whitespace at end of a line.
+
+    Trailing space is all whitespace found after the last non-whitespace
+    character on the line until the newline. This includes tabs "\\\\t",
+    blank lines, blanks etc.
+    """
+    class docs:
+        example = """
+        def func( a ):
+              pass
+
+        """.replace('\n', '\t\n')
+        example_language = 'Python 3'
+        importance_reason = """
+        Trailing spaces make code less readable and maintainable.
+        """
+        fix_suggestions = """
+        Removing the trailing spaces.
+        """
+    allow_trailing_spaces = Taste[bool](
+        'Determines whether or not trailing spaces should be allowed or not.',
+        (True, False), default=False)
+
+
+@Spacing.subaspect
+class BlankLine:
+    """
+    A line with zero characters.
+    """
+    class docs:
+        example = """
+        name = input('What is your name?')
+
+
+        print('Hi, {}'.format(name))
+        """
+        example_language = 'Python 3'
+        importance_reason = """
+        Various programming styles use blank lines in different places.
+        The usage of blank lines affects the readability, maintainability and
+        length of a code i.e blank lines can either make code longer, less
+        readable and maintainable or do the reverse.
+        """
+        fix_suggestions = """
+        Following specific rules about the usage of blank lines: using them
+        only when necessary.
+        """
+
+
+@BlankLine.subaspect
+class BlankLineAfterDeclaration:
+    """
+    Those found after declarations.
+    """
+    class docs:
+        example = """
+        int main ()
+        {
+          int a;
+          float b;
+
+          scanf("%d%f", &a, &b);
+          printf("a = %d and b = %f", a, b);
+          return 0;
+        }
+        """
+        example_language = 'C'
+        importance_reason = """
+        Having a specific and reasonable number of blank lines after every
+        block of declarations improves on the readability of the code.
+        """
+        fix_suggestions = """
+        `BlankLintAfterDeclaration` issues can be fixed specifying (and of
+        course using) a reasonable number of blank lines to use after block
+        declaration.
+        """
+    blank_lines_after_declarations = Taste[int](
+        'Represents the number of blank lines after declarations',
+        (0, 1, 2), default=0)
+
+
+@BlankLine.subaspect
+class BlankLineAfterProcedure:
+    """
+    Those found after procedures or functions.
+    """
+    class docs:
+        example = """
+        void proc(void){
+            printf("this does nothing");
+        }
+
+        int add(float a, float b){
```

Comment:
this example looks like a 'good' example, not a 'bad' example.  Normally these examples include bad examples.

Also example needs `#include`s to be even half valid.